src/main/java/io/english/service/UserService.java

```diff
@@ -36,13 +40,18 @@ public User getById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException(String.format("User not found by id=%d", id)));
     }
 
+    public User getCurrentUser() {
+        Long userId = principalUtils.getUserIdFromPrincipal();
```

Comment:
principalUtils is more a presentation layer object. it should be used only in controller layer. 
In controller you should get currentUserId and then call service method using this id as an argument